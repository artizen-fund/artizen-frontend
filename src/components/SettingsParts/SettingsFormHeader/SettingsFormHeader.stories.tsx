import SettingsFormHeader from './'

export default {
  title: 'components/SettingsFormHeader',
  component: SettingsFormHeader,
  argTypes: {},
}

export const SettingsFormHeaderComponent = (props: any) => <SettingsFormHeader {...props} />
