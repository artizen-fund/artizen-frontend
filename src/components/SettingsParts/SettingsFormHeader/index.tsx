import { useEffect } from 'react'
import styled from 'styled-components'
import { rgba, assetPath } from '@lib'
import { palette, typography } from '@theme'

interface ISettingsFormHeader {
  title: string
  subtitle: string
  imgPath: string
}

const SettingsFormHeader = ({ title, subtitle, imgPath }: ISettingsFormHeader) => (
  <Wrapper>
    <img src={`${assetPath(imgPath)}?fm=webp`} alt="header image" />
    <Title>{title}</Title>
    <Subtitle>{subtitle}</Subtitle>
  </Wrapper>
)

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 10px;
  padding: 48px;
  margin-bottom: 48px;
  border-bottom: 1px solid ${rgba(palette.stone)};

  img {
    max-width: 100%;
  }
  text-align: center;
`

const Title = styled.h2`
  ${typography.title.l2}
`

const Subtitle = styled.p`
  color: ${rgba(palette.barracuda)};
  ${typography.label.l1}
`

export default SettingsFormHeader
