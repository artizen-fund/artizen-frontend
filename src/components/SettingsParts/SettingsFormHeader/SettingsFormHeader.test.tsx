import renderer from 'react-test-renderer'
import SettingsFormHeader from './'

describe('SettingsFormHeader', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SettingsFormHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
