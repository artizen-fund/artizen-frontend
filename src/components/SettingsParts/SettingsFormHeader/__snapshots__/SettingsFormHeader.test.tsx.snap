// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsFormHeader renders correctly 1`] = `
<div
  className="sc-5bd1e25b-0 ePxCsU"
>
  <img
    alt="header image"
    className="sc-5bd1e25b-3 hAmsGH"
    src="undefined?fm=webp"
  />
  <img
    alt="header image"
    className="sc-5bd1e25b-3 erireH"
    src="undefined?fm=webp"
  />
  <h2
    className="sc-5bd1e25b-1 dcUTNt"
  />
  <p
    className="sc-5bd1e25b-2 gMnDiO"
  />
</div>
`;
